platform/viewer/src/routes/Mode/Mode.tsx

```diff
@@ -352,7 +353,9 @@ export default function ModeRoute({
             studyInstanceUIDs[0] !== undefined &&
             renderLayoutData({
               ...layoutTemplateData.current.props,
-              ViewportGridComp: ViewportGridWithDataSource,
+              ViewportGridComp: ViewportGridWithDataSourceFactory(
+                layoutTemplateData.current?.props?.viewportGridProps
```

Comment:
I don't think this file should be touched. proprs that are returned from modes are layout specific, and all of them are passed to the layout specified (currently we have only one layout that accepts leftPanels, rightPanels). You can just use this viewportGridProps there